command/main.go

```diff
@@ -264,3 +253,55 @@ func printCommand(w io.Writer, name string, cmdFn cli.CommandFactory) {
 	}
 	fmt.Fprintf(w, "    %s\t%s\n", name, cmd.Synopsis())
 }
+
+func generateCurlString(exitCode int, runOpts *RunOptions, preParsingErrCatcher *bytes.Buffer) int {
```

Comment:
It might be more idiomatic to specify an interface here as an input (e.g. `preParsingErrCatcher io.Writer`). However, if I understand correctly, this error is supposed to be discarded anyway for `if outputCurlString || outputPolicy {` cases. In this case, would it make sense to just remove this variable altogether?

Reply:
I'm passing it as a *bytes.Buffer because I specifically needed the `WriteTo` method.

```
		fmt.Fprint(preParsingErrCatcher, "cURL command not set by API operation\n")
		preParsingErrCatcher.WriteTo(runOpts.Stderr)
```

This is me writing whatever was caught in that buffer (the purpose of which I explain in my other comment) to runOpts.Stderr. If you know a way to do those two lines with just io.Writer.Write lemme know, I'm not super familiar with the various ways to work with buffers so I just went with the first one that I was able to figure out. 😄

Reply:
I think I can understand the logic a bit better now. In this case, would it be a bit more clear to remove the following block:

```diff
- if outputCurlString || outputPolicy {
-	uiErrWriter = &bytes.Buffer{}
- }
```

then change the `preParsingErrCatcher` to `io.Writer`, and change the lines where we write the errors to just write directly into `preParsingErrCatcher`?

```diff
  fmt.Fprint(preParsingErrCatcher, "cURL command not set by API operation\n")
- preParsingErrCatcher.WriteTo(runOpts.Stderr)
```

If a temporary buffer is required, I would define it within the scope of `generateCurlString`.

Reply:
(We resolved this offline but just for posterity), this buffer needs to be defined in the same scope as it is now so it can be passed to the initialization of the CLI commands, then populated inside the command's Run() function, and finally written to stderr in generateCurlString.

It's adding to an already confusing and hacky part of the codebase so we're interested in having this part of both output-curl-string and output-policy in a future refactoring PR, likely as part of the effort to move this kind of functionality out of the `api` package.